src/main/java/io/pravega/connectors/flink/FlinkPravegaWriter.java

```diff
@@ -288,6 +293,10 @@ protected void commit(PravegaTransactionState transaction) {
                     }
                 } catch (TxnFailedException e) {
                     log.error("{} - Transaction {} commit failed.", writerId(), txn.getTxnId());
+                } catch (RuntimeException e) {
+                    if (e.getMessage().contains("Unknown transaction")) {
+                        log.error("{} - Transaction {} not found.", writerId(), txn.getTxnId());
```

Comment:
Since the `eventStreamWriter` is a mock, is there a way we can test this by doing something like creating a mock that always returns Unknown Transaction?

Reply:
Yes, you are right. I was looking into integration test, and unit tests are easier to have such issue covered. Fixed with the recent commit